tests/api_fastapi/core_api/routes/public/test_connections.py

```diff
@@ -169,3 +169,83 @@ def test_should_respond_200(
+            {
+                "connection_id": "test-connection-id",
+                "conn_type": "test_type",
+                "description": "test_description",
+                "host": "test_host",
+                "login": "test_login",
+                "schema": "test_schema",
+                "port": 8080,
+                "extra": '{"key": "value"}',
+            },
+        ],
+    )
+    def test_post_should_respond_200(self, test_client, session, body):
+        response = test_client.post("/public/connections/", json=body)
+        assert response.status_code == 201
+        connection = session.query(Connection).all()
+        assert len(connection) == 1
+
+    @pytest.mark.parametrize(
+        "body",
+        [
+            {"connection_id": "****", "conn_type": "test_type"},
+            {"connection_id": "test()", "conn_type": "test_type"},
+            {"connection_id": "this_^$#is_invalid", "conn_type": "test_type"},
+            {"connection_id": "iam_not@#$_connection_id", "conn_type": "test_type"},
+        ],
+    )
+    def test_post_should_respond_400_for_invalid_conn_id(self, test_client, body):
+        response = test_client.post("/public/connections/", json=body)
+        assert response.status_code == 400
+        connection_id = body["connection_id"]
+        assert response.json() == {
+            "detail": f"The key '{connection_id}' has to be made of "
+            "alphanumeric characters, dashes, dots and underscores exclusively",
+        }
+
+    @pytest.mark.parametrize(
+        "body",
+        [
+            {"connection_id": "test-connection-id", "conn_type": "test_type"},
+        ],
+    )
+    def test_post_should_respond_already_exist(self, test_client, body):
+        response = test_client.post("/public/connections/", json=body)
+        assert response.status_code == 201
+        # Another request
+        response = test_client.post("/public/connections/", json=body)
+        assert response.status_code == 409
+        assert response.json() == {
+            "detail": "Connection with connection_id: `test-connection-id` already exists",
+        }
+
+    @pytest.mark.parametrize(
+        "body",
+        [
+            {"connection_id": "test-connection-id", "conn_type": "test_type", "password": "test-password"},
+            {"connection_id": "test-connection-id", "conn_type": "test_type", "password": "?>@#+!_%()#"},
+            {
+                "connection_id": "test-connection-id",
+                "conn_type": "test_type",
+                "password": "A!rF|0wi$aw3s0m3",
+                "extra": '{"password": "test-password"}',
+            },
+        ],
+    )
+    def test_post_should_response_201_redacted_password(self, test_client, body):
+        response = test_client.post("/public/connections/", json=body)
+        assert response.status_code == 201
+        connection = response.json()
+        assert "password" not in connection
```

Comment:
This test is not correct. This will only assert that "password" key is not in the response dict this is not enough. We actually want password and `extra` in the response, just the values should be redacted.

You need to check that the response has `"password": "***"` and `"extra": {"password": "***"}`

Also you might be missing a `@pytest.mark.enable_redact` in your test.

Cf `variables`.

Reply:
I first included it like that will return `***` as redacted using similar approaches to the `variables` and `extra` field `connections`. Then I  saw it is not even included in the reponse in the legacy api document. 
[post_connection](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_connection). In the Patch PR comment, I know you have said we should reduct the returned `password`, but the document made me confused, and I pushed this to align with you first. I can easily get back to returning the password field and redacting.

Yeah, for the redacted unit test, indeed this should be missing that marking definition. Thanks for pointing out!

Reply:
One more thing to consider is including a new field to response data model doesn't seem like a breaking change since it can be ignored easily since no one is using the field due to not exist in the response, but it could be for a couple of users if they are tightly coupled with the response schema somehow and persisting somewhere the response.

In any case, maybe we should document this somewhere, either in the common api changes issue or so. What do you think?

Reply:
I think we should add the password, indeed adding a field is not a breaking change and this is how we have been updating the legacy API.

Reply:
Included the password to the response as redacted and adjusted the tests accordingly. Followed up a similar approach with variables. I thi. I will apply the same update to the patch. Many thanks! :)